src/execution/values.js

```diff
@@ -64,9 +67,12 @@ export function getVariableValues(
         ),
       );
     } else {
+      const hasValue = hasOwnProperty(inputs, varName);
       const value = inputs[varName];
-      if (isInvalid(value)) {
+      if (!hasValue || isNullish(value)) {
         if (isNonNullType(varType)) {
+          // If no value or a nullish value was provided to a variable with a
+          // non-null type (required), produce an error.
           errors.push(
             new GraphQLError(
```

Comment:
I think it not confusing to get such error if you pass `null`:
> Variable "${name}" of required type "${String(argType)}" was not provided.

I think it should have a special error message for this case, similar to how it's done for arguments:
> `Variable "${name}" of non-null type "${String(argType)}" must not be null.'

Reply:
Great point, I agree